AdvancedExample/create.ps1

```diff
@@ -0,0 +1,355 @@
+                if ($null -ne $errorObjectConverted.error.message) {
+                    $httpErrorObj.FriendlyMessage = $errorObjectConverted.error.message
+                    if ($null -ne $errorObjectConverted.error.code) { 
+                        $httpErrorObj.FriendlyMessage = $httpErrorObj.FriendlyMessage + " Error code: $($errorObjectConverted.error.code)"
+                    }
+                }
+                else {
+                    $httpErrorObj.FriendlyMessage = $errorObjectConverted.error
+                }
+            }
+            else {
+                $httpErrorObj.FriendlyMessage = $ErrorObject
+            }
+        }
+        catch {
+            $httpErrorObj.FriendlyMessage = $httpErrorObj.ErrorDetails
+        }
+        Write-Output $httpErrorObj
+    }
+}
+
+function New-TeamsSession {
+    [CmdletBinding()]
+    [OutputType([System.Collections.Generic.Dictionary[[String], [String]]])]
+    param(
+        [parameter(Mandatory)]
+        [string]
+        $TenantId,
+
+        [parameter(Mandatory)]
+        [string]
+        $ClientId,
+
+        [parameter(Mandatory)]
+        [string]
+        $ClientSecret
+    )
+    try {
+        $baseUri = "https://login.microsoftonline.com/"
+        $authUri = $baseUri + "$TenantId/oauth2/v2.0/token"
+
+        Write-Verbose "Creating Graph Access Token"
+        $bodyGraph = @{
+            grant_type    = "client_credentials"
+            client_id     = "$ClientId"
+            client_secret = "$ClientSecret"
+            scope      = "https://graph.microsoft.com/.default"
+        }
+        $responseGraph = Invoke-RestMethod -Method POST -Uri $authUri -Body $bodyGraph -ContentType 'application/x-www-form-urlencoded' -Verbose:$false -UseBasicParsing:$true -ErrorAction "Stop"
+        $accessTokenGraph = $responseGraph.access_token
+        Write-Verbose "Succesfully created Graph Access Token"
+
+        Write-Verbose "Creating MS Teams Access Token"
+        $bodyTeams = @{
+            grant_type    = "client_credentials"
+            client_id     = "$ClientId"
+            client_secret = "$ClientSecret"
+            scope      = "48ac35b8-9aa8-4d74-927d-1f4a14a0b239/.default"
+        }
+
+        $responseTeams= Invoke-RestMethod -Method POST -Uri $authUri -Body $bodyTeams -ContentType 'application/x-www-form-urlencoded'-Verbose:$false -UseBasicParsing:$true -ErrorAction "Stop" 
+        $accessTokenTeams = $responseTeams.access_token
+        Write-Verbose "Succesfully created MS Teams Access Token"
+
+        Write-Verbose "Connecting to Microsoft Teams"
+
+        $connectTeamsSplatParams = @{
+            AccessTokens = @("$accessTokenGraph", "$accessTokenTeams")
+            Verbose      = $false
+            ErrorAction  = "Stop"
+        }
+        $teamsSession = Connect-MicrosoftTeams @connectTeamsSplatParams
+        Write-Verbose "Successfully connected to Microsoft Teams"
+
+        Write-Output $teamsSession
```

Comment:
Moeten we deze expliciet ook teruggeven? gezien we de variable nergens gebruiken vermoed ik dat, als we niets teruggeven of; evt op regel 200 iets doen als: `$null = New-TeamsSession` dat we er ook zijn. Dat maakt de code iets duidelijker.

Reply:
pak ik op

Reply:
opgepakt in code